pennylane/ops/qutrit/matrix_ops.py

```diff
@@ -0,0 +1,90 @@
+import warnings
+import numpy as np
+
+import pennylane as qml
+from pennylane.operation import AnyWires, Operation, DecompositionUndefinedError
+from pennylane.wires import Wires
+
+
+class QutritUnitary(Operation):
+    r"""
+    Apply and arbitrary fixed unitary matrix.
+
+    **Details:**
+
+    * Number of wires: Any (the operation can act on any number of wires)
+    * Number of parameters: 1
+    * Gradient recipe: None
+
+    Args:
+        U (array[complex]): square unitary matrix
+        wires(Sequence[int] or int): the wire(s) the operation acts on
+    """
+    num_wires = AnyWires
+    """int: Number of wires that the operator acts on."""
+
+    num_params = 1
+    """int: Number of trainable parameters that the operator depends on."""
+
+    grad_method = None
+    """Gradient computation method."""
+
+    def __init__(self, *params, wires, do_queue=True):
+        wires = Wires(wires)
+
+        # For pure QutritUnitary operations (not controlled), check that the number
+        # of wires fits the dimensions of the matrix
+        U = params[0]
+
+        dim = 3 ** len(wires)
+
+        if qml.math.shape(U) != (dim, dim):
+            raise ValueError(
+                f"Input unitary must be of shape {(dim, dim)} to act on {len(wires)} wires."
+            )
+
+        # Check for unitarity; due to variable precision across the different ML frameworks,
+        # here we issue a warning to check the operation, instead of raising an error outright.
+        if not qml.math.is_abstract(U) and not qml.math.allclose(
+            qml.math.dot(U, qml.math.T(qml.math.conj(U))),
+            qml.math.eye(qml.math.shape(U)[0]),
+            atol=1e-6,
+        ):
+            warnings.warn(
+                f"Operator {U}\n may not be unitary."
+                "Verify unitarity of operation, or use a datatype with increased precision.",
+                UserWarning,
+            )
+
+        super().__init__(*params, wires=wires, do_queue=do_queue)
+
+    @staticmethod
+    def compute_matrix(U):
+        r"""Representation of the operator as a canonical matrix in the computational basis (static method).
+
+        The canonical matrix is the textbook matrix representation that does not consider wires.
+        Implicitly, this assumes that the wires of the operator correspond to the global wire order.
+
+        Args:
+            U (tensor_like): unitary matrix
+
+        Returns:
+            tensor_like: canonical matrix
+        """
+        return U
+
+    @staticmethod
+    def compute_decomposition(U, wires):
+        raise DecompositionUndefinedError
+
+    def adjoint(self):
+        U = self.matrix()
+        return QutritUnitary(qml.math.moveaxis(qml.math.conj(U), -2, -1), wires=self.wires)
```

Comment:
Using `qml.math.moveaxis` with -2, -1 for the axes is the same as transposing the last two axes of a matrix, so in the case of a 2D array it behaves the same way as a matrix transpose. I don't entirely understand why this function is used instead of `qml.math.T`, but it behaves the same way regardless.